docs/adr/2023-09-08_026-stateless-observation-construction.md

```diff
@@ -0,0 +1,138 @@
+---
+slug: 26
+title: |
+  26. Stateless transaction observaion & construction
+authors: []
+tags: [Accepted]
+---
+
+## Status
+
+Accepted
+
+## Context
+
+- [ADR 18](/adr/18) merged both `headState` and `chainState` into one single
+  state in the Hydra node, giving the chain layer a way to _fetch_ and update
+  the `chainState` when observing a chain event.
+
+- [ADR 23](/adr/23) outlined the need for a local chain state in the chain layer
+  again to correctly handle correct observation of multiple relevant
+  transactions and the resulting `chainState` updates.
+
+- The `ChainStateType tx` for our "actual" Cardano chain layer is currently:
+
+  ```haskell
+  data ChainStateAt = ChainStateAt
+    { chainState :: ChainState
+    , recordedAt :: Maybe ChainPoint
+    }
+
+  data ChainState
+    = Idle
+    | Initial InitialState
+    | Open OpenState
+    | Closed ClosedState
+  ```
+
+  where `InitialState`, `OpenState` and `ClosedState` hold elaborate information
+  about the currently tracked Hydra head.
+
+- We face [difficulties](https://github.com/input-output-hk/hydra/issues/529) to
+  provide sufficient user feedback when an `initTx` was observed but (for
+  example) keys do not match our expectation.
+
+  - Core problem is, that `observeInit` is required to take a decision whether
+    it wants to "adopt" the Head by returning an `InitialState` or not.
+  - This makes it impossible to provider user feedback through the `HeadLogic`
+    and `API` layers.
+
+- We want to build a [Hydra head
+  explorer](https://github.com/input-output-hk/hydra/issues/696), which should
+  be able to keep track and discover Hydra heads and their state changes even
+  when the heads were initialzed before starting the explorer.
+
+## Decision
+
+- We supersede [ADR 23](/adr/18) with the current ADR.
+
+### Changes internal to Direct chain layer
+
+- Introduce a `ResolvedTx` type that has its inputs resolved.
```

Comment:
I thought this is a transaction and I guess all txs have their inputs resolved anyway right? 
Instead this type holds also the input utxos as well as the tx itself. Maybe mention the structure of this type to be more clear.

Reply:
Clarified this in the ADR